framework/doc/content/syntax/Functors/index.md

```diff
@@ -0,0 +1,138 @@
+# Functor system
+
+Functors are an abstraction, existing as a base class, that is available to several systems in MOOSE:
+
+- [Variables](syntax/Variables/index.md)
+- [Auxiliary variables](syntax/AuxVariables/index.md)
+- [Functor material properties](syntax/FunctorMaterials/index.md)
+- [Functions](syntax/Functions/index.md)
+
+All functors can be called using the same interfaces. This enables considerable code re-use.
+For example, instead of having a kernel for each type of coupled forcing term, like
+[CoupledForce.md], [BodyForce.md], [MatCoupledForce.md], [ADMatCoupledForce.md], we could just have a single object
+`FunctorForce` and have the force term be a functor.
+
+`Functions` provide a good analogy to `Functors`. `Functions` are evaluated on-the-fly at a location in space and time.
+`Functors` are evaluated on-the-fly with a space and a time argument. The space arguments can be an element, a point in an element,
+a face of an element, and so on. The time arguments represent the state of the functor : current, previous value (whether in time or iteration),
+or value before that previous one.
+
+!alert note
+If a Functor is reported as missing by the simulation, you may use the [!param](/Debug/SetupDebugAction/show_functors)
+to get more information about which functors were created and requested.
+
+## Developing with functors
+
+Functors are evaluated on-the-fly. E.g. they can be viewed as functions of the current location in
+space (and time). Functors provide several overloads of the
+`operator()` method for different "geometric quantities". One example of a
+"geometric quantity" is based around an element, e.g. for an `FVElementalKernel`, the
+value of a functor material property in a cell-averaged sense can be obtained by
+the syntax
+
+- `_foo(makeElemArg(_current_elem), determineState())`
```

Comment:
Before this, you should talk about declaring the member variable (and what type).

Reply:
You're right. I need to move more docs from FunctorMaterials.